pkg/pluginhost/handler_querydata.go

```diff
@@ -29,7 +29,11 @@ func (ds *DataSource) QueryData(ctx context.Context, req *backend.QueryDataReque
 			span.RecordError(err)
 			logger.Error("error un-marshaling the query", "error", err.Error())
 			// Here we are using error source from the original error and if it does not have any source we are using the plugin error as the default source
-			errorRes := backend.ErrorResponseWithErrorSource(backend.PluginError(fmt.Errorf("%s: %w", "error un-marshaling the query", err)))
+			err = fmt.Errorf("%s: %w", "error un-marshaling the query", err)
```

Comment:
@ivanahuckova does this make sense? Is this the way to do this?

Reply:
What is the goal here? If we just want to return error with original source, then the default error source is plugin, so we don't need to add it in a case error does not have a source. 

```
errorRes := backend.ErrorResponseWithErrorSource(fmt.Errorf("%s: %w", "error un-marshaling the query", err))
```

Reply:
nice yeah I think that should be it.